test/contracts/LSP.spec.ts

```diff
@@ -293,4 +299,73 @@ describe("LSP", function () {
     const expectedCollateralBalance = oldCollateralBalance.add(tokensToRedeem)
     expect(newCollateralBalance).toEqBN(expectedCollateralBalance)
   })
+  it("Should change state to ExpiredPriceRequested after expire", async function () {
+    // Check that sponsor still has enough L / S Tokens
+    const syntheticBalances = await contracts.LSP.getPositionTokens(
+      namedAccounts.deployer
+    )
+    expect(syntheticBalances[0]).toBeGteBN(tokensToSettle)
+    expect(syntheticBalances[1]).toBeGteBN(tokensToSettle)
+
+    // Check contract state
+    const ContractState = {
+      Open: 0,
+      ExpiredPriceRequested: 1,
+      ExpiredPriceReceived: 2,
+    }
+
+    let contractState: number = await contracts.LSP.contractState()
+
+    expect(contractState).toEqual(ContractState.Open)
+
+    // fast forward 350 seconds
+    await ethers.provider.send("evm_increaseTime", [350])
+    await ethers.provider.send("evm_mine", [])
+
+    // expire the contract
+    const expireTx = await contracts.LSP.expire()
+
+    await expireTx.wait()
+    contractState = await contracts.LSP.contractState()
+
+    expect(contractState).toEqual(ContractState.ExpiredPriceRequested)
+  })
+
+  it("Should change the contract state after calling expire", async function () {
+    // Check that sponsor still has enough L / S Tokens
+    const syntheticBalances = await contracts.LSP.getPositionTokens(
+      namedAccounts.deployer
+    )
+    expect(syntheticBalances[0]).toBeGteBN(tokensToSettle)
+    expect(syntheticBalances[1]).toBeGteBN(tokensToSettle)
+
+    // Check contract state
+    enum ContractState {
+      Open,
+      ExpiredPriceRequested,
+      ExpiredPriceReceived,
+    }
+
+    let contractState: number = await contracts.LSP.contractState()
+
+    expect(contractState).toEqual(ContractState.Open)
+
+    // fast forward 350 seconds
+    await ethers.provider.send("evm_increaseTime", [350])
+    await ethers.provider.send("evm_mine", [])
+
+    // expire the contract
+    const expireTx = await contracts.LSP.expire()
+
+    await expireTx.wait()
+    contractState = await contracts.LSP.contractState()
+
+    expect(contractState).toEqual(ContractState.ExpiredPriceRequested)
+  })
+  it("Should settle tokens after receiving expired price. ", async () => {
+    contracts.OptimisticOracle = await ethers.getContractAt(
```

Comment:
Having read until here I realise  that this PR is probably still a work in progress / draft.
Would be cool to mark it as such so that I don't jump the gun on reviewing next time 😉

Reply:
Oh whoops, it totally should have been. I appreciate the feedback so far though!